src/DataValues/DecimalMath.php

```diff
@@ -252,115 +252,62 @@ private function roundDigits( $value, $significantDigits ) {
 			return '+0';
 		}
 
+		$len = strlen( $value );
+
+		// keeping all digits means no rounding
+		if ( $significantDigits >= ( $len -1 ) ) {
```

Comment:
How many "significant digits" does `"1230"` have? Should be 3, right? Which is equal to the string length minus one.

What if the passed `$value` is `"1234"` but `$significantDigits` is 3?

Why minus one?